pkg/services/pluginsintegration/clientmiddleware/caching_middleware.go

```diff
@@ -50,12 +55,23 @@ type CachingMiddleware struct {
 	caching  caching.CachingService
 	log      log.Logger
 	features featuremgmt.FeatureToggles
+	tracer   trace.Tracer
 }
 
 // QueryData receives a data request and attempts to access results already stored in the cache for that request.
 // If data is found, it will return it immediately. Otherwise, it will perform the queries as usual, then write the response to the cache.
 // If the cache service is implemented, we capture the request duration as a metric. The service is expected to write any response headers.
 func (m *CachingMiddleware) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	ctx, span := m.tracer.Start(ctx, "CachingMiddleware.QueryData")
+	defer span.End()
+
+	if req != nil {
+		span.SetAttributes(attribute.String("plugin_id", req.PluginContext.PluginID))
+		if ds := req.PluginContext.DataSourceInstanceSettings; ds != nil {
+			span.SetAttributes(attribute.String("datasource_type", ds.Type))
```

Comment:
Just checking if we need this attribute as well since plugin id and datasource type are synonymous